supervisor/resolution/checks/dhcp.py

```diff
@@ -0,0 +1,62 @@
+"""Helpers to check core security."""
+from ...const import CoreState
+from ...coresys import CoreSys
+from ...dbus.const import InterfaceMethod
+from ...dbus.network.interface import NetworkInterface
+from ..const import ContextType, IssueType
+from .base import CheckBase
+
+
+def setup(coresys: CoreSys) -> CheckBase:
+    """Check setup function."""
+    return CheckDHCP(coresys)
+
+
+class CheckDHCP(CheckBase):
+    """CheckDHCP class for check."""
+
+    async def run_check(self) -> None:
+        """Run check if not affected by issue."""
+        for interface in self.sys_dbus.network.interfaces.values():
+            if CheckDHCP.check_interface(interface):
+                self.sys_resolution.create_issue(
+                    IssueType.DHCP_FAILURE,
+                    ContextType.SYSTEM,
+                    interface.name,
+                )
+
+    async def approve_check(self, reference: str | None = None) -> bool:
+        """Approve check if it is affected by issue."""
+        interface = self.sys_dbus.network.interfaces.get(reference)
+
+        return interface and CheckDHCP.check_interface(interface)
+
+    @staticmethod
+    def check_interface(interface: NetworkInterface) -> bool:
+        """Return true if a managed interface has a DHCP issue."""
+        if not (interface.managed and interface.settings):
+            return False
+
```

Comment:
While it looks good to me, I am worried that this doesn't capture some state we don't foresee (static not applying? I don't see how that could happen, but who knows). Did you consider looking at the NetworkManager state flags?

https://developer-old.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMActivationStateFlags

The logic would go something like:

```
            interface.settings.ipv4
            and interface.settings.ipv4.method != InterfaceMethod.DISABLED.value
            StateFlag & NM_ACTIVATION_STATE_FLAG_IP4_READY
```

Reply:
StateFlag is actually on the Active Connection info. It also considers ip6 and ip4 ready if they are disabled since it has nothing to do there. So the check is simpler now, basically just consider it a network connection error if the state flag doesn't have these two values in it regardless of the method used.

I also added another check of `State` not being in `Activated` or `Activating` as that's a problem too. Supervisor provides no way to deactivate an existing interface so if a managed interface isn't in one of those states then there's an issue the user needs to look at.